gui/src/components/BridgeForm.js

```diff
@@ -0,0 +1,89 @@
+import React, { Fragment } from 'react'
+import { withFormik, Form } from 'formik'
+import { withStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import { TextField, Grid } from '@material-ui/core'
+import { postBridge } from 'api'
+
+const styles = theme => ({
+  textfield: {
+    paddingTop: theme.spacing.unit * 1.25
+  },
+  form: {
+    paddingTop: theme.spacing.unit * 4
+  },
+  card: {
+    paddingBottom: theme.spacing.unit * 2
+  },
+  button: {
+    marginTop: theme.spacing.unit * 3
+  }
+})
+
+const FormLayout = ({ isSubmitting, classes, handleChange }) => (
+  <Fragment>
+    <Form className={classes.form} noValidate>
+      <Grid container justify='center' spacing={0}>
+        <Grid item xs={2}>
+          <TextField
+            fullWidth
+            onChange={handleChange}
+            className={classes.textfield}
+            label='Type Bridge Name'
+            type='name'
+            name='name'
+            placeholder='name'
+          />
+        </Grid>
+      </Grid>
+      <Grid container justify='center' spacing={0}>
+        <Grid item xs={2}>
+          <TextField
+            label='Type Bridge URL'
+            type='url'
+            name='url'
+            placeholder='url'
+            fullWidth
+            onChange={handleChange}
+            className={classes.textfield}
+          />
+        </Grid>
+      </Grid>
+      <Grid container justify='center' spacing={0}>
+        <Grid item xs={2}>
+          <TextField
+            onChange={handleChange}
+            className={classes.textfield}
+            fullWidth
+            type='confirmations'
+            name='confirmations'
+            placeholder='confirmations'
+            label='Type Confirmations'
+          />
+        </Grid>
+      </Grid>
+      <Grid container justify='center' spacing={0}>
+        <Button color='primary' type='submit' className={classes.button} disabled={isSubmitting}>
+          Build Bridge
+        </Button>
+      </Grid>
+    </Form>
+  </Fragment>
+)
+
+const BridgeForm = withFormik({
+  mapPropsToValues ({ name, url, confirmations }) {
+    return {
+      name: name || '',
+      url: url || '',
+      confirmations: confirmations || ''
+    }
+  },
+  handleSubmit (values) {
+    const formattedValues = JSON.parse(JSON.stringify(values).replace('confirmations', 'defaultConfirmations'))
+    formattedValues.defaultConfirmations = parseInt(formattedValues.defaultConfirmations) || 0
+    postBridge(formattedValues)
```

Comment:
You are calling `postBridge` directly on `api.js` rather than through redux in `action.js`. Any particular reason for breaking the paradigm set with the `fetch*` actions?

Reply:
@NavyAdmiral agreed that we want to send these through redux actions. There's some upcoming work on a global loading indicator that will leverage it.